baseplate/context/memcache.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from pymemcache.client.base import PooledClient
+
+from . import ContextFactory
+from .. import config
+
+
+def pool_from_config(app_config, prefix="memcache.", **kwargs):
+    """Make a PooledClient from a configuration dictionary.
+
+    The keys useful to :py:func:`pool_from_config` should be prefixed, e.g.
+    ``memcache.host``, ``memcache.max_pool_size``, etc. The ``prefix`` argument
+    specifies the prefix used to filter keys. Each key is mapped to a
+    corresponding keyword argument on the
+    `PooledClient <https://pymemcache.readthedocs.io/en/latest/apidoc/pymemcache.client.base.html#pymemcache.client.base.PooledClient>`_
+    constructor.
+
+    Supported keys:
+
+    * ``host`` (required): a string representing an IP or domain to connect
+        to memcached, e.g. ``127.0.0.1`` or ``localhost``.
+    * ``port`` (required): an integer representing the port to connect to memcached
+        for the given host, e.g. ``11211``.
+    * ``max_pool_size``: an integer for the maximum pool size to use, by default
+        this is ``2147483648``.
+    * ``connect_timeout``: a float representing seconds to wait for a connection to
+        memcached server. Defaults to the underlying socket default timeout.
+    * ``timeout``: a float representing seconds to wait for calls on the
+        socket connected to memcache. Defaults to the underlying socket default
+        timeout.
+
+    """
+
+    assert prefix.endswith(".")
+    config_prefix = prefix[:-1]
+    cfg = config.parse_config(app_config, {
+        config_prefix: {
+            "host": config.String,
```

Comment:
it'd be more consistent with other parts of baseplate if you did 

```python
"endpoint": config.Endpoint,
```

then you can just do `cfg.endpoint.address` as the first parameter of `PooledClient`